### ISSUE TITLE
No matching distribution found for tensorflow==2.4.1

### ISSUE THREAD

**Issue by zohimchandani** [tensorflow/tensorflow#53709]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MAC OS Big Sur 11.6
- TensorFlow installed from (source or binary): via pip 
- TensorFlow version: trying to install 2.4.1
- Python version: 3.9.6
- Installed using virtualenv? pip? conda?: pip 

**Describe the problem**

Via jupyter notebooks on vs code, trying to run `!pip3 install tensorflow==2.4.1` but getting 
```
ERROR: Could not find a version that satisfies the requirement tensorflow==2.4.1 (from versions: 2.5.0rc0, 2.5.0rc1, 2.5.0rc2, 2.5.0rc3, 2.5.0, 2.5.1, 2.5.2, 2.6.0rc0, 2.6.0rc1, 2.6.0rc2, 2.6.0, 2.6.1, 2.6.2, 2.7.0rc0, 2.7.0rc1, 2.7.0, 2.8.0rc0)
ERROR: No matching distribution found for tensorflow==2.4.1
```

**Comment by sushreebarsa** [comment]
@zohimchandani Could you please try to latest stable `TF v2.7.0` and refer to [build from source](https://www.tensorflow.org/install/source#macos), [tested build configurations](https://www.tensorflow.org/install/source#macos).Please have a look at the  similar [issue1](https://github.com/tensorflow/tensorflow/issues/47205), [issue2](https://github.com/tensorflow/tensorflow/issues/43526)and  [system requirements](https://www.tensorflow.org/install/pip#system-requirements) .Please let us know if it helps?Thank you!

**Comment by bguetarni** [comment]
This is because you are trying to install tensorflow 2.4 with python 3.9. As you see [here](https://www.tensorflow.org/install/source_windows#gpu), tensorflow 2.4 is only supported with python 3.6 / 3.7 / 3.8 .

Creates an environment with one of these python versions and then you could install tensorflow 2.4 .

**Comment by trish118** [comment]
I am getting the same error could anyone give me detailed information on how to actually do it

**Comment by yoursaptarshi** [comment]
I am also getting same error

**Comment by sushreebarsa** [comment]
@trish118 @yoursaptarshi Reopening this ticket as the issue still persists.  
Thank you!

**Comment by 652994331** [comment]
@bguetarni hi, I am trying to install tensorflow==2.4.0   with python3.8, I got error like this: 
Python 3.8.13 (default, Mar 28 2022, 06:13:39) 
[Clang 12.0.0 ] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
exit>>> exit()
(test) hqz@hqzs-MacBook-Air Documents % pip show tensorflow
WARNING: Package(s) not found: tensorflow
(test) hqz@hqzs-MacBook-Air Documents % pip install tensorflow==2.4.0
ERROR: Could not find a version that satisfies the requirement tensorflow==2.4.0 (from versions: 2.13.0rc0, 2.13.0rc1, 2.13.0rc2, 2.13.0)
ERROR: No matching distribution found for tensorflow==2.4.0
(test) hqz@hqzs-MacBook-Air Documents % pip install tensorflow==2.2.0
ERROR: Could not find a version that satisfies the requirement tensorflow==2.2.0 (from versions: 2.13.0rc0, 2.13.0rc1, 2.13.0rc2, 2.13.0)
ERROR: No matching distribution found for tensorflow==2.2.0

**Comment by bguetarni** [comment]
On my machine, with Windows 10, when I try your command `pip install tensorflow==2.2.0` with an environment that has python 3.8, it works fine. I suspect that your environment **test** has not python 3.8 installed but another version. Can you do `python --version` within your environment please.

In your error, it shows the available tensorflow versions (2.13.0rc0, 2.13.0rc1, 2.13.0rc2, 2.13.0) which indicates that you probably have python 3.11.

**Comment by sushreebarsa** [comment]
Please try to check the python version you have been using in your environment as it seems the incompatible python version with the TF version. Could you please try with the compatible one. You can use the following command to use any specific version and check this [link](https://www.tensorflow.org/install/source_windows#gpu) to see the compatible configuration. 
```
!pip3 install tensorflow==tf_version
```
Thank you!
